src/libexpr/meson.build

```diff
@@ -0,0 +1,164 @@
+
+
+# src files
+#============================================================================
+
+libexpr_src = files(
+    'attr-path.cc',
+    'attr-set.cc',
+    'common-eval-args.cc',
+    'eval-cache.cc',
+    'eval.cc',
+    'function-trace.cc',
+    'get-drvs.cc',
+    'json-to-value.cc',
+    'nixexpr.cc',
+    'primops.cc',
+    'value-to-json.cc',
+    'value-to-xml.cc')
+
+libexpr_headers = files(
+    'attr-path.hh',
+    'attr-set.hh',
+    'common-eval-args.hh',
+    'eval-cache.hh',
+    'eval.hh',
+    'eval-inline.hh',
+    'function-trace.hh',
+    'get-drvs.hh',
+    'json-to-value.hh',
+    'nixexpr.hh',
+    'primops.hh',
+    'symbol-table.hh',
+    'value.hh',
+    'value-to-json.hh',
+    'value-to-xml.hh')
+
+
+# include directories
+#========================================================================
+
+# include current dir
+#---------------------------------------------------
+libexpr_inc = [include_directories('.')]
+
+# add subdirectories
+#---------------------------------------------------
+libexpr_dirs = [
+  'flake',
+  'primops']
+
+foreach dir : libexpr_dirs
+    subdir(dir)
+endforeach
+
+
+# targets
+#============================================================================
+
+libexpr_src += custom_target(
+    'parser_tab.[cchh]',
+    output : [
+        'parser-tab.cc',
+        'parser-tab.hh'],
+    input : 'parser.y',
+    command : [
+        bison,
+        '-v',
+        '--output=@OUTPUT0@',
+        '--defines=@OUTPUT1@',
+        '@INPUT@'])
+
+libexpr_src += custom_target(
+    'lexer_tab.[cchh]',
+    output : ['lexer-tab.cc', 'lexer-tab.hh'],
+    input : 'lexer.l',
+    command : [
+        flex,
+        '--outfile=@OUTPUT0@',
+        '--header-file=@OUTPUT1@',
+        '@INPUT@'])
+
+libexpr_src += custom_target(
+  'imported-drv-to-derivation.nix.gen.hh',
+  output : 'imported-drv-to-derivation.nix.gen.hh',
+  input : 'imported-drv-to-derivation.nix',
+  command : [bash, '-c', gen_rheader, 'sh', '@OUTPUT@'])
+
+
+libexpr_src += custom_target(
+  'fetchurl.nix.gen.hh',
+  output : 'fetchurl.nix.gen.hh',
+  input : 'fetchurl.nix',
+  command : [bash, '-c', gen_rheader, 'sh', '@OUTPUT@'])
+
+
+# build
+#============================================================================
+
+# set build args
+#---------------------------------------------------
+libexpr_cxx_args = []
+
+libexpr_link_args = []
+
+
+# build library
+#---------------------------------------------------
+libexpr_lib = library(
+    'nixexpr',
+    sources : libexpr_src,
+    install : true,
+    install_mode : 'rwxr-xr-x',
+    install_dir : libdir,
+    include_directories : [
+        libexpr_inc,
+        proj_inc,
+    ],
+    cpp_args : libexpr_cxx_args,
+    link_args : libexpr_link_args,
+    dependencies : libexpr_dep_list)
+
+# install headers
+#---------------------------------------------------
+install_headers(
+    libexpr_headers,
+    install_dir : join_paths(includedir, 'nix'))
```

Comment:
I think this can just be `'nix'` assuming includedir is equivalent to `get_option('includedir')`

Reply:
Well no, install_dir is prefix-relative.

But you could specify the subdir as `'nix'` instead, which is a kwarg specific to this function and specifically meant to avoid needing to handle the includedir.

Reply:
Nice catch